src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java

```diff
@@ -358,6 +395,73 @@ public void afterRebuildExecution( IncrementalBuildHelperRequest incrementalBuil
 
     }
 
+    private void writeChangedFiles(
+            DirectoryScanner directoryScanner,
+            String[] filesBeforeAction,
+            File mojoConfigBase,
+            String createdFilesLstFilename ) throws MojoExecutionException
+    {
+        if ( directoryScanner.getBasedir() == null )
+        {
+            return;
+        }
+
+        DirectoryScanResult outputDirectoryScanResult = scanDirectoryDiff(
+                directoryScanner, filesBeforeAction );
+
+        try
+        {
+            writeChangedFiles( outputDirectoryScanResult, mojoConfigBase, createdFilesLstFilename );
+        }
+        catch ( IOException e )
+        {
+            throw new MojoExecutionException( "Error while storing the mojo status", e );
+        }
+    }
+
+    private static void writeChangedFiles(
+            DirectoryScanResult outputDirectoryScanResult,
+            File mojoConfigBase,
+            String createdFilesLstFilename )
+            throws IOException
+    {
+        File createdFiles = new File( mojoConfigBase, createdFilesLstFilename );
+        FileUtils.fileWriteArray( createdFiles, outputDirectoryScanResult.getFilesAdded() );
```

Comment:
This method is deprecated. Use

java.nio.files.Files.write(file.toPath(), data.getBytes(encoding), StandardOpenOption.CREATE)

and specify UTF-8 encoding